DOCUMENTATION.md

```diff
@@ -43,6 +45,190 @@ The overall territory is laid out here in this [C4](https://c4model.com) System
 
 ![Propulsion c4model.com Container Diagram: Reactors](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.github.com/jet/propulsion/master/diagrams/ReactorsContainer.puml&fmt=svg)
 
+# The `Propulsion.Streams` Programming Model for Projections, Reactions and Workflows
+
+`Propulsion.Streams` provides a programming model to manage running of _Handlers_ in a manner that optimises for the following:
+- isolating the handlers from the client libraries of a given Event Store (the Handler is triggered via a 'Sink' that manages accepts incoming events and checkpointing of progress in a Store-specific manner relevant to the hosting environment in which your handler will be run).
+- providing a clean approach to the testing of Reaction logic with and without involving your actual Event Store (the `MemoryStoreProjector` component is a key part of that story).
+- exposing thorough diagnostic information that enables one to effectively monitor and tune your Handlers in the context of an overall production system.
+
+## Overview of running projections with Propulsion
+
+### Glossary
+
+#### Tranches
```

Comment:
```suggestion
#### Optimised processing with Tranches
```

Reply:
Hm, that was a definition of a term more than an attempt to claim it was anything new or special. Had a to/fro with @jeffijoe on the Discord about using Partition as the term for that (I once used it when it was Cosmos only, but I feel it might think about the wrong things, especially for feeds, where the elements might be warehouses, tables etc).

Reply:
I haven't yet seen `tranches` used in this context in another tooling. It's a good name, but if you just present it, people may skip it while skimming. Shard could be also a good name imho.

Reply:
Marking this resolved:
- DynamoStore Index's Tranches have been corrected to be termed Partitions
- Glossary now defines Tranches
- Difference between Tranches vs Partitions (you can run multiple tranches over a single partition) means another term is needed

Thanks for calling it out - I know it's now a complete wall of text, but that's better than having to explain Tranches in the context of its use every time.